node-backend-starter-code/public/index.html

```diff
@@ -0,0 +1,330 @@
+              if (!searchText) {
+                  alert('Please enter movie name to search for');
+                  searchTextField.focus();
+                  return false;
+              }
+
+              findMovies(searchText);
+
+              return false; // prevent form from submitting and refreshing entire page
+          };
+
+          // Executed when Favorites button is clicked
+          document.getElementById('listFavorites').onclick = function () {
+              var apiUrl = window.location.origin + '/favorites',
+                  data = JSON.stringify({}),
+                  callbackParams = [];
+
+              ajaxCall('GET', apiUrl, data, true, favoritesCallback, callbackParams);
+          };
+
+          /*** PUBLIC METHODS ***/
+
+          /**
+           * Find movies by title
+           *
+           * @param  string title
+           * @param  int    page  Page number for search results which may be split into multiple pages
+           * @return void
+           */
+          function findMovies(title, page)
+          {
+              // HTTPS needed for API url especially if deployed on Heroku which uses HTTPS
+              var page = parseInt(page) || 1,
+                  apiUrl = 'https://www.omdbapi.com/?r=json&page=' + page + '&s='
+                         + encodeURIComponent(title).replace(/%20/, '+'),
+                  callbackParams = {"title":title, "page":page};
+
+              ajaxCall('GET', apiUrl, '', false, findMoviesCallback, callbackParams);
+          }
+
+          /**
+           * Save movie to favorites
+           *
+           * @param  int resultIndex
+           * @return void
+           */
+          function saveMovie(resultIndex)
+          {
+              var movie = getMovie(resultIndex),
+                  apiUrl = window.location.origin + '/favorites',
+                  callbackParams = [],
+                  data;
+
+              if (false === movie) {
+                  alert('Unable to save movie to Favorites');
+                  return;
+              }
+
+              data = JSON.stringify({"name":movie.Title, "oid":movie.imdbID});
+              ajaxCall('POST', apiUrl, data, true, saveMovieCallback, callbackParams);
+          }
+
+          /**
+           * Show movie details
+           *
+           * @param  int resultIndex
+           * @return void
+           */
+          function showDetails(resultIndex)
+          {
+              var movie = getMovie(resultIndex),
+                  html = '';
+
+              if (false === movie) {
+                  detailsBox.innerHTML = 'No details';
+                  return;
+              }
+
+              html += '<h2>Movie Details</h2>';
+
+              for (var property in movie) {
+                  if (movie.hasOwnProperty(property)) { // exclude prototype properties
+                      html += '<b>' + property + ':</b> ' + movie[property] + '<br>';
```

Comment:
Yes, especially with the data coming from an external API. Suggested fix would be to escape on movie[property].